tutorials/early_stopping/early_stopping.ipynb

```diff
@@ -0,0 +1,1067 @@
+   "id": "2fec7495",
+   "metadata": {},
+   "source": [
+    "## Setting up the SearchSpace\n",
+    "\n",
+    "First, we define our search space. Ax supports both range parameters\n",
+    "of type integer and float as well as choice parameters which can have\n",
+    "non-numerical types such as strings.\n",
+    "We will tune the hidden sizes, learning rate, and dropout parameters."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "cf6f869f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "parameters = [\n",
+    "    # NOTE: In a real-world setting, hidden_size_1 and hidden_size_2\n",
+    "    # should probably be powers of 2, but in our simple example this\n",
+    "    # would mean that num_params can't take on that many values, which\n",
+    "    # in turn makes the Pareto frontier look pretty weird.\n",
+    "    RangeParameter(\n",
+    "        name=\"hidden_size_1\",\n",
+    "        lower=16,\n",
+    "        upper=128,\n",
+    "        parameter_type=ParameterType.INT,\n",
+    "        log_scale=True,\n",
+    "    ),\n",
+    "    RangeParameter(\n",
+    "        name=\"hidden_size_2\",\n",
+    "        lower=16,\n",
+    "        upper=128,\n",
+    "        parameter_type=ParameterType.INT,\n",
+    "        log_scale=True,\n",
+    "    ),\n",
+    "    RangeParameter(\n",
+    "        name=\"learning_rate\",\n",
+    "        lower=1e-4,\n",
+    "        upper=1e-2,\n",
+    "        parameter_type=ParameterType.FLOAT,\n",
+    "        log_scale=True,\n",
+    "    ),\n",
+    "    RangeParameter(\n",
+    "        name=\"dropout\",\n",
+    "        lower=0.0,\n",
+    "        upper=0.5,\n",
+    "        parameter_type=ParameterType.FLOAT,\n",
+    "    ),\n",
+    "]\n",
+    "\n",
+    "search_space = SearchSpace(\n",
+    "    parameters=parameters,\n",
+    "    # NOTE: In practice, it may make sense to add a constraint\n",
+    "    # hidden_size_2 <= hidden_size_1\n",
+    "    parameter_constraints=[],\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "a8005e80",
+   "metadata": {},
+   "source": [
+    "## Setting up Metrics\n",
+    "\n",
+    "Ax has the concept of a Metric that defines properties of outcomes and how observations are obtained for these outcomes. This allows e.g. encodig how data is fetched from some distributed execution backend and post-processed before being passed as input to Ax.\n",
+    "\n",
+    "We will optimize the validation accuracy, which is a `TensorboardCurveMetric` that points to the logging directory assigned above. Note that we have set `is_available_while_running`, allowing for the metric to be queried as the trial progresses. This is critical for the early stopping strategy to monitor partial results."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "0775a96e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MyTensorboardMetric(TensorboardCurveMetric):\n",
```

Comment:
I wonder if it makes sense to check in sth like a `LocalDirTensorBoardCurveMetric` that can be instantiated with the log dir so that this can just be imported rather than needing to be defined.